datafusion/core/src/sql/planner.rs

```diff
@@ -2968,6 +3016,28 @@ mod tests {
         );
     }
 
+    #[test]
+    fn select_array_no_common_type() {
+        let sql = "SELECT [1, true, null]";
+        let err = logical_plan(sql).expect_err("query should have failed");
+
+        // HashSet doesnt guaranty order
+        assert_contains!(
+            format!("{:?}", err),
+            r#"NotImplemented("Arrays with different types are not supported: "#
+        );
```

Comment:
```suggestion
        // HashSet doesn't guarantee order
        assert_contains!(
            err.to_string(),
            r#"Arrays with different types are not supported: "#
        );
```

Reply:
Just a suggestion -- what you have here is fine too

Reply:
👍 Thanks